src/main/java/sg/ncl/MainController.java

```diff
@@ -2560,12 +2560,130 @@ public String rejectTeam(
         return "redirect:/admin";
     }
 
+    @RequestMapping("/admin/teams/{teamId}")
+    public String restrictFreeTeams(
+            @PathVariable final String teamId,
+            @RequestParam(value = "action", required=true) final String action,
+            final RedirectAttributes redirectAttributes,
+            HttpSession session) throws IOException
+    {
+        // check if admin
+        if (!validateIfAdmin(session)) {
+            log.warn("Access denied when trying to restrict/free team {}: must be admin!", session.getAttribute(webProperties.getSessionUserId()));
+            return NO_PERMISSION_PAGE;
+        }
+
+        Team2 team = invokeAndExtractTeamInfo(teamId);
+
+        // check if team is approved before restricted
+        if ("restrict".equals(action) && team.getStatus().equals(TeamStatus.APPROVED.name())) {
+            return restrictTeam(team, redirectAttributes);
+        }
+        // check if team is restricted before freeing it back to approved
+        else if ("free".equals(action) && team.getStatus().equals(TeamStatus.RESTRICTED.name())) {
+            return freeTeam(team, redirectAttributes);
+        } else {
+            log.warn("Error in restrict/free team {}: failed to {} team with status {}", teamId, action, team.getStatus());
+            redirectAttributes.addFlashAttribute(MESSAGE, ERROR_PREFIX + "failed to " + action + " team " + team.getName() + " with status " + team.getStatus());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal " team " 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)